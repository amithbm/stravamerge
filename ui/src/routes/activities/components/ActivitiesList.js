import React from 'react'
import ActivityListItem from './ActivityListItem'
import {ButtonGroup, Button, Glyphicon} from 'react-bootstrap' 

export default ({activities, canMoveNext, canMovePrevious, moveNext, movePrevious}) =>
    <div>
        <ButtonGroup className="pull-right" bsSize="small">
            <Button disabled={!canMovePrevious} onClick={canMovePrevious ? movePrevious : undefined}><Glyphicon glyph="chevron-left" /></Button>
            <Button disabled={!canMoveNext} onClick={canMoveNext ? moveNext : undefined}><Glyphicon glyph="chevron-right" /></Button>
        </ButtonGroup>
        <h3 className="m-t-0">Activities</h3>
        {activities.map((activity,index) =>
            <ActivityListItem activity={activity} key={index} includeHyperlinks={true} />
        )}
    </div>
