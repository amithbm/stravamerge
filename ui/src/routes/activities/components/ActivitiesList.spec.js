import React from 'react'
import {mount, shallow} from 'enzyme'
import renderer from 'react-test-renderer'
import ActivitiesList from './ActivitiesList'
import { getActivitiesTestData } from '../../../testing/testData'
import { Button } from 'react-bootstrap' 

describe('components', () => {
    describe('<ActivitiesList />', () => {
        it('renders as expected', () => {
            const activities = getActivitiesTestData().slice(0,5)
            const component =  renderer.create(<ActivitiesList
                activities={activities}
                canMoveNext={true}
                canMovePrevious={true}
                moveNext={() => 0}
                movePrevious={() => 0} />)
            const tree = component.toJSON()
            expect(tree).toMatchSnapshot()
        })

        it('calls move next when canMoveNext is true', () => {
            let eventFired = false
            const activities = getActivitiesTestData().slice(0,5)
            const component = mount(<ActivitiesList
                activities={[]}
                canMoveNext={true}
                moveNext={() => eventFired=true}
                 />)
            component.find(Button).last().simulate('click')
            expect(eventFired).toBe(true)
        })

        it('does not call move next when canMoveNext is false', () => {
            let eventFired = false
            const component = mount(<ActivitiesList
                activities={[]}
                canMoveNext={false}
                moveNext={() => eventFired=true}
                 />)
            component.find(Button).last().simulate('click')
            expect(eventFired).toBe(false)
        })

        it('calls move previous when canMovePrevious is true', () => {
            let eventFired = false
            const component = mount(<ActivitiesList
                activities={[]}
                canMovePrevious={true}
                movePrevious={() => eventFired=true}
                 />)
            component.find(Button).first().simulate('click')
            expect(eventFired).toBe(true)
        })

        it('does not call move previous when canMovePrevious is false', () => {
            let eventFired = false
            const component = mount(<ActivitiesList
                activities={[]}
                canMovePrevious={false}
                movePrevious={() => eventFired=true}
                 />)
            component.find(Button).first().simulate('click')
            expect(eventFired).toBe(false)
        })
    })
})