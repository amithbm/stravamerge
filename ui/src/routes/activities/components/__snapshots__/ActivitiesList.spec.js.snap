exports[`components <ActivitiesList /> renders as expected 1`] = `
<div>
  <div
    className="pull-right btn-group btn-group-sm">
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      <span
        className="glyphicon glyphicon-chevron-left" />
    </button>
    <button
      className="btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button">
      <span
        className="glyphicon glyphicon-chevron-right" />
    </button>
  </div>
  <h3
    className="m-t-0">
    Activities
  </h3>
  <div
    className="m-b-3">
    <div
      className="row">
      <div
        className="col-xs-12">
        <p
          className="m-b">
          <span
            className="title-label">
            Test 1
          </span>
          <span
            className="subtitle-label">
             - 
            Fri Jan 01 2016 (Ride)
          </span>
        </p>
        <table>
          <tbody>
            <tr>
              <td
                width="0">
                <span
                  className="m-r">
                  <em>
                    Distance
                  </em>
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  0
                   km
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  <em>
                    Elapsed Time
                  </em>
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  5
                   minutes
                </span>
              </td>
              <td
                className="m-r-2">
                <span
                  className="m-r">
                  <em>
                    Type
                  </em>
                </span>
              </td>
              <td>
                Ride
              </td>
            </tr>
            <tr>
              <td
                width="0">
                <span
                  className="m-r">
                  <em>
                    Avg Speed
                  </em>
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  NaN
                   kph
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  <em>
                    Max Speed
                  </em>
                </span>
              </td>
              <td>
                34
                 kph
              </td>
              <td
                className="m-r-2">
                <span
                  className="m-r">
                  <em>
                    Trainer
                  </em>
                </span>
              </td>
              <td>
                Yes
              </td>
            </tr>
          </tbody>
        </table>
        <div
          className="m-t">
          <a
            className="m-r-2"
            onClick={[Function]}
            style={Object {}}>
            Compare Segments
          </a>
          <a
            className="m-r-2"
            onClick={[Function]}
            style={Object {}}>
            3D Route Profile
          </a>
        </div>
      </div>
    </div>
    <hr />
  </div>
  <div
    className="m-b-3">
    <div
      className="row">
      <div
        className="col-xs-12">
        <p
          className="m-b">
          <span
            className="title-label">
            Test 2
          </span>
          <span
            className="subtitle-label">
             - 
            Sat Jan 02 2016 (Ride)
          </span>
        </p>
        <table>
          <tbody>
            <tr>
              <td
                width="0">
                <span
                  className="m-r">
                  <em>
                    Distance
                  </em>
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  0
                   km
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  <em>
                    Elapsed Time
                  </em>
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  5
                   minutes
                </span>
              </td>
              <td
                className="m-r-2">
                <span
                  className="m-r">
                  <em>
                    Type
                  </em>
                </span>
              </td>
              <td>
                Ride
              </td>
            </tr>
            <tr>
              <td
                width="0">
                <span
                  className="m-r">
                  <em>
                    Avg Speed
                  </em>
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  NaN
                   kph
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  <em>
                    Max Speed
                  </em>
                </span>
              </td>
              <td>
                34
                 kph
              </td>
              <td
                className="m-r-2">
                <span
                  className="m-r">
                  <em>
                    Trainer
                  </em>
                </span>
              </td>
              <td>
                Yes
              </td>
            </tr>
          </tbody>
        </table>
        <div
          className="m-t">
          <a
            className="m-r-2"
            onClick={[Function]}
            style={Object {}}>
            Compare Segments
          </a>
          <a
            className="m-r-2"
            onClick={[Function]}
            style={Object {}}>
            3D Route Profile
          </a>
        </div>
      </div>
    </div>
    <hr />
  </div>
  <div
    className="m-b-3">
    <div
      className="row">
      <div
        className="col-xs-12">
        <p
          className="m-b">
          <span
            className="title-label">
            Test 3
          </span>
          <span
            className="subtitle-label">
             - 
            Mon Feb 01 2016 (Ride)
          </span>
        </p>
        <table>
          <tbody>
            <tr>
              <td
                width="0">
                <span
                  className="m-r">
                  <em>
                    Distance
                  </em>
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  0
                   km
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  <em>
                    Elapsed Time
                  </em>
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  5
                   minutes
                </span>
              </td>
              <td
                className="m-r-2">
                <span
                  className="m-r">
                  <em>
                    Type
                  </em>
                </span>
              </td>
              <td>
                Ride
              </td>
            </tr>
            <tr>
              <td
                width="0">
                <span
                  className="m-r">
                  <em>
                    Avg Speed
                  </em>
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  NaN
                   kph
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  <em>
                    Max Speed
                  </em>
                </span>
              </td>
              <td>
                34
                 kph
              </td>
              <td
                className="m-r-2">
                <span
                  className="m-r">
                  <em>
                    Trainer
                  </em>
                </span>
              </td>
              <td>
                Yes
              </td>
            </tr>
          </tbody>
        </table>
        <div
          className="m-t">
          <a
            className="m-r-2"
            onClick={[Function]}
            style={Object {}}>
            Compare Segments
          </a>
          <a
            className="m-r-2"
            onClick={[Function]}
            style={Object {}}>
            3D Route Profile
          </a>
        </div>
      </div>
    </div>
    <hr />
  </div>
  <div
    className="m-b-3">
    <div
      className="row">
      <div
        className="col-xs-12">
        <p
          className="m-b">
          <span
            className="title-label">
            Test 4
          </span>
          <span
            className="subtitle-label">
             - 
            Tue Feb 02 2016 (Ride)
          </span>
        </p>
        <table>
          <tbody>
            <tr>
              <td
                width="0">
                <span
                  className="m-r">
                  <em>
                    Distance
                  </em>
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  0
                   km
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  <em>
                    Elapsed Time
                  </em>
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  5
                   minutes
                </span>
              </td>
              <td
                className="m-r-2">
                <span
                  className="m-r">
                  <em>
                    Type
                  </em>
                </span>
              </td>
              <td>
                Ride
              </td>
            </tr>
            <tr>
              <td
                width="0">
                <span
                  className="m-r">
                  <em>
                    Avg Speed
                  </em>
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  NaN
                   kph
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  <em>
                    Max Speed
                  </em>
                </span>
              </td>
              <td>
                34
                 kph
              </td>
              <td
                className="m-r-2">
                <span
                  className="m-r">
                  <em>
                    Trainer
                  </em>
                </span>
              </td>
              <td>
                Yes
              </td>
            </tr>
          </tbody>
        </table>
        <div
          className="m-t">
          <a
            className="m-r-2"
            onClick={[Function]}
            style={Object {}}>
            Compare Segments
          </a>
          <a
            className="m-r-2"
            onClick={[Function]}
            style={Object {}}>
            3D Route Profile
          </a>
        </div>
      </div>
    </div>
    <hr />
  </div>
  <div
    className="m-b-3">
    <div
      className="row">
      <div
        className="col-xs-12">
        <p
          className="m-b">
          <span
            className="title-label">
            Test 5
          </span>
          <span
            className="subtitle-label">
             - 
            Tue Mar 01 2016 (Ride)
          </span>
        </p>
        <table>
          <tbody>
            <tr>
              <td
                width="0">
                <span
                  className="m-r">
                  <em>
                    Distance
                  </em>
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  0
                   km
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  <em>
                    Elapsed Time
                  </em>
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  5
                   minutes
                </span>
              </td>
              <td
                className="m-r-2">
                <span
                  className="m-r">
                  <em>
                    Type
                  </em>
                </span>
              </td>
              <td>
                Ride
              </td>
            </tr>
            <tr>
              <td
                width="0">
                <span
                  className="m-r">
                  <em>
                    Avg Speed
                  </em>
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  NaN
                   kph
                </span>
              </td>
              <td
                width="0">
                <span
                  className="m-r">
                  <em>
                    Max Speed
                  </em>
                </span>
              </td>
              <td>
                34
                 kph
              </td>
              <td
                className="m-r-2">
                <span
                  className="m-r">
                  <em>
                    Trainer
                  </em>
                </span>
              </td>
              <td>
                Yes
              </td>
            </tr>
          </tbody>
        </table>
        <div
          className="m-t">
          <a
            className="m-r-2"
            onClick={[Function]}
            style={Object {}}>
            Compare Segments
          </a>
          <a
            className="m-r-2"
            onClick={[Function]}
            style={Object {}}>
            3D Route Profile
          </a>
        </div>
      </div>
    </div>
    <hr />
  </div>
</div>
`;
